staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -3507,9 +3520,14 @@ func (c *Cloud) buildELBSecurityGroupList(serviceName types.NamespacedName, load
 	var err error
 	var securityGroupID string
 
+	sgList := getSGListFromAnnotation(annotations[ServiceAnnotationLoadBalancerSecurityGroups])
+
+	// The below code changes makes sure that when we have Security Groups  specified with the ServiceAnnotationLoadBalancerSecurityGroups
+	// annotation we don't create a new default Security Groups
+
```

Comment:
nit, while whole part can be reorganized to like to make the logic more clear. But current one is ok too.
```
if len(sgList) == 0 {
    var securityGroupID string
    if c.cfg.Global.ElbSecurityGroup != "" {
       ...
    } else {
        ...
    }
    sgList = append(sgList, securityGroupID)
}
```

Reply:
Good point , Thanks for the tip.